server.go

```diff
@@ -445,13 +448,12 @@ func (s *server) peerConnected(conn net.Conn, connReq *connmgr.ConnReq, inbound
 
 	// Now that we've established a connection, create a peer, and
 	// it to the set of currently active peers.
-	p, err := newPeer(conn, connReq, s, peerAddr, inbound)
+	p, err := newPeer(conn, connReq, s, peerAddr, inbound, s.chanDB)
 	if err != nil {
 		srvrLog.Errorf("unable to create peer %v", err)
-		if p.connReq != nil {
-			s.connMgr.Remove(p.connReq.ID())
+		if connReq != nil {
+			s.connMgr.Remove(connReq.ID())
 		}
-		p.Disconnect()
```

Comment:
If we're unable to create the peer, then it must be removed from the `connmgr`'s set of pending persistent connections, hence the use of `Disconnect` here. 

Honestly, we need to revisit the current `connmgr` integration for coherency as it was put together rather quickly in order to get the functionality out the door.